python/chemiscope/jupyter.py

```diff
@@ -146,7 +153,9 @@ def show_input(path, mode="default", warning_timeout=10000):
     elif mode == "map":
         widget_class = MapWidget
 
-    if path.endswith(".gz"):
+    if isinstance(path, io.IOBase):
```

Comment:
I'm not sure this is the best way to check, aren't there file-like objects that don't inherit from IOBase? I would rather do the check the other way around:

```py
if isinstance(file, (str, Path)):
    # current code
else:
    # assume file-like object
```

Reply:
Changed this to check for `str` or `Path` first.